common/models/types/build.sh

```diff
@@ -10,101 +10,63 @@ set -eu
 # adjust relative paths as necessary
 THIS_SCRIPT="$(readlink -f "${BASH_SOURCE[0]}")"
 . "${THIS_SCRIPT%/*}/../../../resources/build/build-utils.sh"
-. "$KEYMAN_ROOT/resources/shellHelperFunctions.sh"
 ## END STANDARD BUILD SCRIPT INCLUDE
 
-EX_USAGE=64
-
-
-display_usage ( ) {
-  echo "Usage: $0 [-test] [-publish-to-npm]"
-  echo "       $0 -help"
-  echo
-  echo "  -help               displays this screen and exits"
-  echo "  -test               runs tests"
-  echo "  -publish-to-npm     publishes the current version to the npm package index"
-  echo "  -dry-run            do test, etc, but don't actually publish"
-}
-
-################################ Main script ################################
-
-run_tests=0
-install_dependencies=1
-should_publish=0
-npm_dist_tag=
-should_dry_run=0
-
-# Process command-line arguments
-while [[ $# -gt 0 ]] ; do
-  key="$1"
-  case $key in
-    -help|-h)
-      display_usage
-      exit
-      ;;
-    -dry-run)
-      should_dry_run=1
-      ;;
-    -test)
-      run_tests=1
-      install_dependencies=0
-      ;;
-    -version)
-      echo "Warning: -version is now ignored"
-      ;;
-    -tier)
-      echo "Warning: -tier is now ignored"
-      ;;
-    -publish-to-npm)
-      should_publish=1
-      ;;
-    *)
-      echo "$0: invalid option: $key"
-      display_usage
-      exit $EX_USAGE
-  esac
-  shift # past the processed argument
-done
-
-# Dry run settings
-if (( should_dry_run )); then
-  DRY_RUN=--dry-run
-else
-  DRY_RUN=
+cd "$THIS_SCRIPT_PATH"
+
+. "$KEYMAN_ROOT/resources/shellHelperFunctions.sh"
+
+
+builder_describe "Build Keyman kmc Package Compiler module" \
+  "@/common/web/keyman-version" \
+  "clean" \
+  "configure" \
+  "build" \
+  "test" \
+  "pack" \
```

Comment:
Can we have a description of pack?  ...mirrors 'publish' for all npm packages

Reply:
Huh, and it usually comes second here, after `publish`... yeah, could definitely use some more clarity there on exactly what the distinction is.

Reply:
"build a local .tgz pack for testing" okay? See #8588, will fix up other refs shortly.